lib/archethic/oracle_chain/services/uco_price/providers/coin_gecko.ex

```diff
@@ -34,7 +34,8 @@ defmodule Archethic.OracleChain.Services.UCOPrice.Providers.Coingecko do
            :httpc.request(:get, {query, []}, httpc_options, []),
          {:ok, payload} <- Jason.decode(body),
          {:ok, prices} <- Map.fetch(payload, "archethic") do
-      {:ok, prices}
+      formatted_prices = Enum.map(prices, fn {pair, price} -> {pair, [price]} end)
+      {:ok, formatted_prices}
```

Comment:
What is the reason of this formatting?

Reply:
because one service returns a list of prices from different sources, so I thought it's better to aggregate the prices at the top level and not on each provider

Reply:
OK I get it.
In terms of aggregation we will have more granularity and precision.
However, we have to think it's the weight of the incorrect values will impact too much the final aggregation

Reply:
we have a trust_level provided by the api, the values are high low and not_provided

Reply:
on one provider, but the best would be to discard any value that has a variance greater than let's say 10%.
WDYT ?

Reply:
but for that we need to trust one source blindly

Reply:
Finally the median of all the values will remove the outlier and not blind them